components/class-m-chart-chartjs.php

```diff
@@ -0,0 +1,328 @@
+		'line'   => 'line',
+	);
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		add_filter( 'm_chart_image_support', array( $this, 'm_chart_image_support'), 10, 2 );
+	}
+
+	/**
+	 * Get the necessary chart data for a given chart and assign it the right class vars
+	 *
+	 * @param int $post_id the WP ID for the chart post
+	 * @param array $args any args we want to override the defaults for
+	 */
+	public function get_chart_data( $post_id, $args ) {
+		$this->args = wp_parse_args( $args, m_chart()->get_chart_default_args );
+		$this->post = get_post( $post_id );
+		$this->post_meta = m_chart()->get_post_meta( $this->post->ID );
+	}
+
+	/**
+	 * Returns an arary of all settings and data needed to build a chart
+	 *
+	 * @param int $post_id WP post ID of the post you want chart args for
+	 * @param array $args any args we want to override the defaults for
+	 * @param bool $force optional param to force a rebuild of the data even if a cache was found
+	 * @param bool $cache optional param to override the default behavior to cache the results
+	 *
+	 * @return string URL to the plugin directory with path if parameter was passed
+	 */
+	public function get_chart_args( $post_id, $args, $force = false, $cache = true ) {
+		// There's a ton of work that goes into generating the chart args so we cache them
+		$cache_key = $post_id . '-chart-args';
+
+		if ( ! $force && $chart_args = wp_cache_get( $cache_key, m_chart()->slug ) ) {
+			// The width can be set via the args so we'll override whatever the cache has with the arg value
+			$chart_args['graph']['width'] = isset( $this->args['width'] ) && is_numeric( $this->args['width'] ) ? $this->args['width'] : '';
+			return $chart_args;
+		}
+
+		if ( ! $this->args || ! $this->post || ! $this->post_meta ) {
+			$this->get_chart_data( $post_id, $args );
+		}
+
+		// Run the parse class on the data
+		m_chart()->parse()->parse_data( $this->post_meta['data']['sets'][0], $this->post_meta['parse_in'] );
+
+		$type = $this->post_meta['type'];
+
+		$chart_args = array(
+			'type' => $this->chart_types[ $this->post_meta['type'] ],
+			'options' => array(
+			    'title' => array(
+			    	'display'   => true,
+					'text'      => $this->esc_title( apply_filters( 'the_title', $this->post->post_title ) ),
+					'fontSize'  => 21,
+					'fontStyle' => 'normal',
+			    ),
+				'legend' => array(
+					'display'  => $this->post_meta['legend'] ? true : false,
+					'position' => 'bottom',
+					'labels' => array(
+						'fontStyle' => 'bold',
+					),
+				),
+				'responsive' => true,
+				'maintainAspectRatio' => false,
+			),
+		);
+
+		$chart_args['data']['labels'] = $this->get_value_labels_array();
+
+		if ( 'pie' != $chart_args['type'] ) {
+			$chart_args = $this->add_axis_labels( $chart_args );
+		}
+
+		if ( 'horizontalBar' == $chart_args['type'] ) {
+			$chart_args['options']['scales']['yAxes'][0]['gridLines']['display'] = false;
+		} else if ( 'pie' != $chart_args['type'] ) {
```

Comment:
There's a few of these and I'm not sure if this is an accident or not, but [PSR-2 convention](https://www.php-fig.org/psr/psr-2/#51-if-elseif-else) (and WordPress convention, too) is `elseif` over `else if`.